service/frontend/handler.go

```diff
@@ -360,11 +384,35 @@ func (wh *WorkflowHandler) PollForDecisionTask(
 
 	wh.Service.GetLogger().Debugf("Poll for decision. DomainName: %v, DomainID: %v", domainName, info.ID)
 
+	pollerID := uuid.New()
 	matchingResp, err := wh.matching.PollForDecisionTask(ctx, &m.PollForDecisionTaskRequest{
 		DomainUUID:  common.StringPtr(info.ID),
+		PollerID:    common.StringPtr(pollerID),
 		PollRequest: pollRequest,
 	})
 	if err != nil {
+		// First check if this err is due to context cancellation.  This means client connection to frontend is closed.
+		if ctx.Err() == context.Canceled {
+			// Our rpc stack does not propagates context cancellation to the other service.  Lets make an explicit
+			// call to matching to notify this poller is gone to prevent any tasks being dispatched to zombie pollers.
+			err = wh.matching.CancelOutstandingPoll(nil, &m.CancelOutstandingPollRequest{
```

Comment:
is it standard to use nil as context if there is no context? or is it better to use context.Background()?

Reply:
updated it to context.Background.